api/users/views.py

```diff
@@ -505,6 +506,31 @@ def get_queryset(self):
         return Node.find(self.get_query_from_request())
 
 
+class UserPreprints(UserNodes):
+    required_read_scopes = [CoreScopes.USERS_READ]
+
+    serializer_class = PreprintSerializer
+    view_category = 'users'
+    view_name = 'user-preprints'
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        user = self.get_user()
+
+        query = (
+            Q('is_deleted', 'ne', True) &
+            Q('contributors', 'eq', user._id) &
+            Q('preprint_file', 'ne', None) &
+            Q('is_public', 'eq', True)
+        )
+
+        return query
+
+    def get_queryset(self):
+        nodes = Node.find(self.get_query_from_request())
+        return [node for node in nodes if node.is_preprint]
```

Comment:
Not a big deal, but this looks inefficient -- evaluating all the nodes takes a lot longer than returning a queryset, and you'll have to do so again to serialize each one. Is it not possible to construct a query guaranteed to return only preprints?